Tanga/Features/Authentication/AppleSignInManager.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  AppleSignInManager.swift
+//  Tanga
+//
+//  Created by Rygel Louv on 26/01/2025.
+//
+
+import Foundation
+import AuthenticationServices
+import CryptoKit
+
+
+class AppleSignInManager: NSObject {
+    
+    static let shared = AppleSignInManager()
+    
+    fileprivate static var currentNonce: String?
+    
+    static var nonce: String? {
+        currentNonce ?? nil
+    }
+    
+    private var continuation : CheckedContinuation<ASAuthorizationAppleIDCredential, Error>?
+    
+    func requestAppleAuthorization() async throws -> ASAuthorizationAppleIDCredential {
+        return try await withCheckedThrowingContinuation { continuation in
```

Comment:
Some guard for potential retain cycles
```suggestion
        return try await withCheckedThrowingContinuation { [weak self] continuation in
        guard let self else { return }
```

Reply:
I don't know very little about retain cycles in swift, could you explain how and why a retain cycle can happen here?

Reply:
Yes sure, 

In the `⁠requestAppleAuthorization()` method, we use ⁠[weak self] in the continuation closure:
```swift
try await withCheckedThrowingContinuation { [weak self] continuation in
    guard let self else { return }
    // ...
}
```

**Why we need this:**
	1.	The continuation closure captures ⁠`self` (`AppleSignInManager` instance)
	2.	The `AppleSignInManager` instance holds a reference to the continuation
	3.	Without ⁠`[weak self]`, this creates a strong reference cycle (memory leak):
	▪	The closure strongly holds onto ⁠self
	▪	⁠self strongly holds onto the continuation
	▪	The continuation strongly holds onto the closure

**Solution:**
By using `⁠[weak self]`:
	•	We break the strong reference cycle
	•	The closure now holds a weak reference to ⁠`self`
	•	If `AppleSignInManager` gets deallocated, the closure won't keep it alive
The ⁠`guard let self else { return }` ensures we have a valid instance to work with and provides a safe early exit if the instance has been deallocated.

Hope it's clear :)

Reply:
I see, I see. So that's how it works?
Thanks for explaining this is valuable feedback.